Common/Subworlds/RavencrestContent/RavencrestSystem.cs

```diff
@@ -0,0 +1,99 @@
+﻿using PathOfTerraria.Common.Systems.Networking.Handlers;
+using PathOfTerraria.Common.Systems.StructureImprovementSystem;
+using SubworldLibrary;
+using System.Collections.Generic;
+using Terraria.DataStructures;
+using Terraria.ID;
+using Terraria.ModLoader.IO;
+
+namespace PathOfTerraria.Common.Subworlds.RavencrestContent;
+
+public class RavencrestSystem : ModSystem
+{
+	private readonly Dictionary<string, ImprovableStructure> structures = [];
+
+	public bool SpawnedScout = false;
+	public Point16 EntrancePosition;
+	public bool ReplacedBuildings = false;
+
+	public override void OnWorldLoad()
+	{
+		ReplacedBuildings = false;
+	}
+
+	public override void PreUpdateTime()
+	{
+		if (Main.netMode != NetmodeID.MultiplayerClient && !ReplacedBuildings && Main.CurrentFrameFlags.ActivePlayersCount > 0)
+		{
+			if (SubworldSystem.Current is RavencrestSubworld)
+			{
+				foreach (ImprovableStructure structure in structures.Values)
+				{
+					structure.Place();
+				}
+			}
+
+			ReplacedBuildings = true;
+		}
+	}
+
+	public override void Load()
+	{
+		structures.Add("Lodge", new ImprovableStructure(2)
+		{
+			StructurePath = "Assets/Structures/RavencrestBuildings/Lodge_",
+			Position = new Point(304, 168),
+		});
+
+		structures.Add("Forge", new ImprovableStructure(2)
+		{
+			StructurePath = "Assets/Structures/RavencrestBuildings/Forge_",
+			Position = new Point(243, 181)
+		});
+	}
+
+	public static void UpgradeBuilding(string name, int level = -1)
+	{
+		if (Main.netMode == NetmodeID.MultiplayerClient)
+		{
+			RavencrestBuildingIndex.Send(name, level);
+		}
+		else
+		{
+			ImprovableStructure structure = ModContent.GetInstance<RavencrestSystem>().structures[name];
+			level = level == -1 ? structure.StructureIndex + 1 : level;
+			structure.Change(level);
+		}
```

Comment:
```suggestion
		if (Main.netMode == NetmodeID.MultiplayerClient)
		{
			RavencrestBuildingIndex.Send(name, level);
			return;
		}
		ImprovableStructure structure = ModContent.GetInstance<RavencrestSystem>().structures[name];
		level = level == -1 ? structure.StructureIndex + 1 : level;
		structure.Change(level);
```

Personal opinion, not mandatory.

Reply:
I do like that more, didn't think of it when writing. Thanks!